Nil map panic in application subcharts when trying to update status field

```
E0224 18:19:01.594696       1 runtime.go:78] Observed a panic: "assignment to entry in nil map" (assignment to entry in nil map)
goroutine 171 [running]:
k8s.io/apimachinery/pkg/util/runtime.logPanic(0x18e2be0, 0x1d70f00)
	/go/pkg/mod/k8s.io/apimachinery@v0.17.5/pkg/util/runtime/runtime.go:74 +0xa3
k8s.io/apimachinery/pkg/util/runtime.HandleCrash(0x0, 0x0, 0x0)
	/go/pkg/mod/k8s.io/apimachinery@v0.17.5/pkg/util/runtime/runtime.go:48 +0x82
panic(0x18e2be0, 0x1d70f00)
	/usr/local/go/src/runtime/panic.go:679 +0x1b2
github.com/Azure/Orkestra/controllers.(*ApplicationGroupReconciler).reconcileApplications(0xc004b230e0, 0x1de47a0, 0xc000954c20, 0xc00084f200, 0x0, 0x0)
	/workspace/controllers/appgroup_reconciler.go:77 +0x4da
github.com/Azure/Orkestra/controllers.(*ApplicationGroupReconciler).reconcile(0xc004b230e0, 0x1dd2e40, 0xc000046030, 0x1de47a0, 0xc000954c20, 0x0, 0x0, 0xc00084f200, 0xc00084f200, 0x0, ...)
	/workspace/controllers/appgroup_reconciler.go:32 +0x2c2
github.com/Azure/Orkestra/controllers.(*ApplicationGroupReconciler).Reconcile(0xc004b230e0, 0x0, 0x0, 0xc004bb2f88, 0x8, 0xc004b50c00, 0x0, 0x0, 0x0)
	/workspace/controllers/appgroup_controller.go:116 +0x749
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler(0xc004a5d680, 0x1947e40, 0xc001226960, 0x43ed00)
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.4.0/pkg/internal/controller/controller.go:256 +0x162
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem(0xc004a5d680, 0x0)
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.4.0/pkg/internal/controller/controller.go:232 +0xcb
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).worker(0xc004a5d680)
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.4.0/pkg/internal/controller/controller.go:211 +0x2b
k8s.io/apimachinery/pkg/util/wait.JitterUntil.func1(0xc004b561b0)
	/go/pkg/mod/k8s.io/apimachinery@v0.17.5/pkg/util/wait/wait.go:152 +0x5e
k8s.io/apimachinery/pkg/util/wait.JitterUntil(0xc004b561b0, 0x3b9aca00, 0x0, 0x1, 0xc0005e67e0)
	/go/pkg/mod/k8s.io/apimachinery@v0.17.5/pkg/util/wait/wait.go:153 +0xf8
k8s.io/apimachinery/pkg/util/wait.Until(0xc004b561b0, 0x3b9aca00, 0xc0005e67e0)
	/go/pkg/mod/k8s.io/apimachinery@v0.17.5/pkg/util/wait/wait.go:88 +0x4d
created by sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func1
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.4.0/pkg/internal/controller/controller.go:193 +0x328
panic: assignment to entry in nil map [recovered]
	panic: assignment to entry in nil map
```

--- a/controllers/appgroup_reconciler.go
+++ b/controllers/appgroup_reconciler.go
@@ -51,6 +51,10 @@ func (r *ApplicationGroupReconciler) reconcileApplications(l logr.Logger, appGro
 		ll := l.WithValues("application", application.Name)
 		ll.V(3).Info("performing chart actions")
 
+		if appGroup.Status.Applications[i].Subcharts == nil {
+			appGroup.Status.Applications[i].Subcharts = make(map[string]orkestrav1alpha1.ChartStatus)
+		}
+
 		repoKey := application.Spec.ChartRepoNickname
 		repoPath := application.Spec.RepoPath
 		name := application.Spec.HelmReleaseSpec.Name

